File: Easy/Rotate Array.py


#  Question Link :: "https://practice.geeksforgeeks.org/problems/rotate-array-by-n-elements-1587115621/1"



# Solution #

class Solution:
    #Function to rotate an array by d elements in counter-clockwise direction. 
    def rotateArr(self,arr,d,n):
        #Your code here
        lst = arr[d:] + arr[0:d]
        
        for i in range(n):
            arr[i] = lst[i]
  

File: Medium/Rotate Array.cpp


// Question Link :: "https://leetcode.com/problems/rotate-array/"


// Solution //

class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        
        int n = nums.size(), i;
        
        if( k <= n ){
            vector<int> temp(n);
            for(i=0;i<k;i++)
                temp[i] = nums[n-k+i];
            for(i; i<n; i++)
                temp[i] = nums[i-k];

            nums=temp;
        }
        else{
            rotate(nums, n);
            rotate(nums, k-n);
        }
    }
};